tests/linalg_test.py

```diff
@@ -2357,5 +2358,59 @@ def testPascal(self, n, kind):
     self._CompileAndCheck(jsp_fun, args_maker)
 
 
+class TestSolveSylvester(jtu.JaxTestCase):
+  # float32 is not close enough for assert
+  jax.config.update("jax_enable_x64", True)  # Ensure float64 for higher precision
+
+  @jtu.sample_product(
+    shape=[(2, 3), (4, 6), (5, 7), (100, 300)],
+  )
+  def test_solve_sylvester(self, shape):
+    n, m = shape
+
+    A = random.normal(random.key(0), shape=(n, n))
+    B = random.normal(random.key(1), shape=(m, m))
+    X_true = random.normal(random.key(2), shape=(n, m))
+
+    C = A @ X_true + X_true @ B
+    with jax.numpy_dtype_promotion('standard'):
+      X_computed = solve_sylvester(A, B, C)
+    assert jnp.allclose(X_computed, X_true)
+
+  @jtu.sample_product(
+    n=[3, 6, 7, 100],
+    dtype = float_types
+  )
+  def test_no_solution_sylvester(self, n, dtype):
+    # Test no solution case to AX + XB = C
+    # For example, the simple case A=[[1]], B = [[-1]], and C = [[1]] would generate (1)X + X(-1) = (1)
+    # which equals X - X = (1) which is not possible because X - X = 0
+    with jax.numpy_dtype_promotion('standard'):
```

Comment:
Same here: we should remove the standard dtype promotion. Everything in this block should execute fine with strict promotion.